### PR TITLE
add io.mpv.Mpv

### PR DESCRIPTION
**Pull request by fastrizwaan**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://github.com/mpv-player/mpv/issues/6933**
- [X ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). **mpv.io** is owned by MPV
- [X ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* **None**

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by PranavBhattarai** [comment]
Really hope this goes well.

**Comment by hfiguiere** [comment]
bot, build io.mpv.Mpv

**Comment by fastrizwaan** [comment]
bot, build io.mpv.Mpv

**Comment by fastrizwaan** [comment]
> Really hope this goes well.

?

File: io.mpv.Mpv.yml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  - "/bin"
+  - "/include"
+  - "/lib/pkgconfig"
+  - "/share/man"
+  sources:
+  - type: archive
+    url: https://www.freedesktop.org/software/uchardet/releases/uchardet-0.0.6.tar.xz
+    sha256: 8351328cdfbcb2432e63938721dd781eb8c11ebc56e3a89d0f84576b96002c61
+
+- name: fribidi
```
Comment (tinywrkb) [inline io.mpv.Mpv.yml:73]: Why? You already have libfribidi in the runtime.

File: io.mpv.Mpv.yml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+finish-args:
+- "--share=ipc"
+- "--socket=x11"
+- "--socket=wayland"
+- "--device=all"
+- "--share=network"
+- "--socket=pulseaudio"
+- "--filesystem=host:ro"
+- "--env=LC_NUMERIC=C"
+- "--env=XDG_DATA_DIRS=/usr/share:/app/share/"
```
Comment (tinywrkb) [inline io.mpv.Mpv.yml:20]: This is wrong. You shouldn't override `XDG_DATA_DIRS`, Flatpak is already setting this correctly. It's possible that you're breaking here fontconfig caching of host-installed fonts, breaking subtitles rendering.

File: io.mpv.Mpv.yml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+
+
+modules:
+
+- udev/udev-175.json
```
Comment (tinywrkb) [inline io.mpv.Mpv.yml:26]: udev? in a Flatpak? why?

**Comment by tinywrkb** [comment]
I would also suggest considering enabling JACK. You'll get this for free when PipeWire will be updated to `0.3.26` in the Flatpak runtime though you'll still need to add socket bind mount.  
Look how I do this [here](https://github.com/tinywrkb/flatpaks/blob/master/shared-modules/pipewire/pipewire-jack-runtime.json). The socket is `- --filesystem=xdg-run/pipewire-0:ro`.

File: io.mpv.Mpv.yml (hunk 3 above)
Comment (tinywrkb) [reply io.mpv.Mpv.yml:26]: also, you have libudev.so in the runtime.

File: io.mpv.Mpv.yml (hunk 2 above)
Comment (fastrizwaan) [reply io.mpv.Mpv.yml]: > This is wrong. You shouldn't override `XDG_DATA_DIRS`, Flatpak is already setting this correctly. It's possible that you're breaking here fontconfig caching of host-installed fonts, breaking subtitles rendering.

https://github.com/fastrizwaan/flathub/commit/37e5f80538f5dd88579b6101075077a332403b09

**Review by fastrizwaan — COMMENTED** [review]
Is appdata ok? Thank you.

**Comment by fastrizwaan** [comment]
> I would also suggest considering enabling JACK. You'll get this for free when PipeWire will be updated to `0.3.26` in the Flatpak runtime though you'll still need to add socket bind mount.
> Look how I do this [here](https://github.com/tinywrkb/flatpaks/blob/master/shared-modules/pipewire/pipewire-jack-runtime.json). The socket is `- --filesystem=xdg-run/pipewire-0:ro`.

Sure, should I copy the `pipewire` directory and as build as shared-module, and  add  `- --filesystem=xdg-run/pipewire-0:ro`? Is `jack` available in runtime?

**Comment by fastrizwaan** [comment]
bot, build io.mpv.Mpv

**Comment by fastrizwaan** [comment]
https://flathub.org/builds/#/builders/32/builds/46801 
looks like `-ljack` is failing in `aarch64`

**Comment by tinywrkb** [comment]
> https://flathub.org/builds/#/builders/32/builds/46801
> https://flathub.org/builds/#/builders/51/builds/3543
> looks like `-ljack` is failing in `aarch64`
> 
> ```
> /usr/lib/gcc/aarch64-unknown-linux-gnu/10.2.0/../../../../aarch64-unknown-linux-gnu/bin/ld: cannot find -ljack
> collect2: error: ld returned 1 exit status
> ```

Right, this is expected as the pkgconfig has arch-specific paths hardcoded. A lazy workaround 

```
diff --git a/shared-modules/pipewire/pipewire-jack-runtime-aarch64.pc b/shared-modules/pipewire/pipewire-jack-runtime-aarch64.pc
new file mode 100644
index 0000000..bb302d1
--- /dev/null
+++ b/shared-modules/pipewire/pipewire-jack-runtime-aarch64.pc
@@ -0,0 +1,11 @@
+prefix=/app
+exec_prefix=/app
+libdir=/usr/lib/aarch64-linux-gnu/pipewire-0.3/jack
+includedir=/app/include
+server_libs=-L/usr/lib/aarch64-linux-gnu/pipewire-0.3/jack -ljackserver
+
+Name: jack
+Description: the Jack Audio Connection Kit: a low-latency synchronous callback-based media server
+Version: 1.9.16
+Libs: -L/usr/lib/aarch64-linux-gnu/pipewire-0.3/jack -ljack
+Cflags: -I/app/include
diff --git a/shared-modules/pipewire/pipewire-jack-runtime.pc b/shared-modules/pipewire/pipewire-jack-runtime-x86_64.pc
similarity index 100%
rename from shared-modules/pipewire/pipewire-jack-runtime.pc
rename to shared-modules/pipewire/pipewire-jack-runtime-x86_64.pc
diff --git a/shared-modules/pipewire/pipewire-jack-runtime.json b/shared-modules/pipewire/pipewire-jack-runtime.json
index ea1ffe6..20a512f 100644
--- a/shared-modules/pipewire/pipewire-jack-runtime.json
+++ b/shared-modules/pipewire/pipewire-jack-runtime.json
@@ -14,7 +14,19 @@
     },
     {
       "type": "file",
-      "path": "pipewire-jack-runtime.pc"
+      "only-arches": [
+        "x86_64"
+      ],
+      "dest-filename": "pipewire-jack-runtime.pc",
+      "path": "pipewire-jack-runtime-x86_64.pc"
+    },
+    {
+      "type": "file",
+      "only-arches": [
+        "aarch-64"
+      ],
+      "dest-filename": "pipewire-jack-runtime.pc",
+      "path": "pipewire-jack-runtime-aarch64.pc"
     }
   ],
   "cleanup": [
~
```

Note that when PipeWire will be bumped to 0.3.26 (now on 0.3.22), the pkgconfig file and JACK headers should be already available in the runtime (PipeWire has new `jack-devel` build option that needs to be enabled).

**Comment by fastrizwaan** [comment]
bot, build io.mpv.Mpv

**Comment by fastrizwaan** [comment]
> Build [46803 successful](https://flathub.org/builds/#/builders/32/builds/46803)
> To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/45019/io.mpv.Mpv.flatpakref
> ```

with jack2 it seems to be build 
 for aarch64, https://github.com/fastrizwaan/flathub/commit/c1d9d24cb28843d266de5a50c2d48635a382f04e

**Comment by fastrizwaan** [comment]
bot, build io.mpv.Mpv

**Comment by fastrizwaan** [comment]
> > https://flathub.org/builds/#/builders/32/builds/46801
> > https://flathub.org/builds/#/builders/51/builds/3543
> > looks like `-ljack` is failing in `aarch64`
> > ```
> > /usr/lib/gcc/aarch64-unknown-linux-gnu/10.2.0/../../../../aarch64-unknown-linux-gnu/bin/ld: cannot find -ljack
> > collect2: error: ld returned 1 exit status
> > ```
> 
> Right, this is expected as the pkgconfig has arch-specific paths hardcoded. A lazy workaround
> 
> ```
> diff --git a/shared-modules/pipewire/pipewire-jack-runtime-aarch64.pc b/shared-modules/pipewire/pipewire-jack-runtime-aarch64.pc
> new file mode 100644
> index 0000000..bb302d1
> --- /dev/null
> +++ b/shared-modules/pipewire/pipewire-jack-runtime-aarch64.pc
> @@ -0,0 +1,11 @@
> +prefix=/app
> +exec_prefix=/app
> +libdir=/usr/lib/aarch64-linux-gnu/pipewire-0.3/jack
> +includedir=/app/include
> +server_libs=-L/usr/lib/aarch64-linux-gnu/pipewire-0.3/jack -ljackserver
> +
> +Name: jack
> +Description: the Jack Audio Connection Kit: a low-latency synchronous callback-based media server
> +Version: 1.9.16
> +Libs: -L/usr/lib/aarch64-linux-gnu/pipewire-0.3/jack -ljack
> +Cflags: -I/app/include
> diff --git a/shared-modules/pipewire/pipewire-jack-runtime.pc b/shared-modules/pipewire/pipewire-jack-runtime-x86_64.pc
> similarity index 100%
> rename from shared-modules/pipewire/pipewire-jack-runtime.pc
> rename to shared-modules/pipewire/pipewire-jack-runtime-x86_64.pc
> diff --git a/shared-modules/pipewire/pipewire-jack-runtime.json b/shared-modules/pipewire/pipewire-jack-runtime.json
> index ea1ffe6..20a512f 100644
> --- a/shared-modules/pipewire/pipewire-jack-runtime.json
> +++ b/shared-modules/pipewire/pipewire-jack-runtime.json
> @@ -14,7 +14,19 @@
>      },
>      {
>        "type": "file",
> -      "path": "pipewire-jack-runtime.pc"
> +      "only-arches": [
> +        "x86_64"
> +      ],
> +      "dest-filename": "pipewire-jack-runtime.pc",
> +      "path": "pipewire-jack-runtime-x86_64.pc"
> +    },
> +    {
> +      "type": "file",
> +      "only-arches": [
> +        "aarch-64"
> +      ],
> +      "dest-filename": "pipewire-jack-runtime.pc",
> +      "path": "pipewire-jack-runtime-aarch64.pc"
>      }
>    ],
>    "cleanup": [
> ~
> ```
> 
> Note that when PipeWire will be bumped to 0.3.26 (now on 0.3.22), the pkgconfig file and JACK headers should be already available in the runtime (PipeWire has new `jack-devel` build option that needs to be enabled).

aarch64 unable to find :
`install: cannot stat 'pipewire-jack-runtime.pc': No such file or directory`
https://flathub.org/builds/#/builders/20/builds/3544

**Comment by fastrizwaan** [comment]
bot, build io.mpv.Mpv

**Comment by tinywrkb** [comment]
```
 "only-arches": [
+        "aarch-64"
+      ],
```

Sorry, that's a typo, should be `aarch64`, not `aarch-64`.

**Comment by fastrizwaan** [comment]
bot, build io.mpv.Mpv

**Comment by fastrizwaan** [comment]
@tinywrkb Congrats!

File: io.mpv.Mpv.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2012-2021 mpv.io -->
+<component type="desktop">
+ <id>io.mpv.Mpv</id>
+ <metadata_license>CC-BY-SA-4.0</metadata_license>
```
Comment (hfiguiere) [inline io.mpv.Mpv.appdata.xml:5]: CC-0 is preferred as a license for the metadata.

**Comment by hfiguiere** [comment]
> Look how I do this [here](https://github.com/tinywrkb/flatpaks/blob/master/shared-modules/pipewire/pipewire-jack-runtime.json). The socket is `- --filesystem=xdg-run/pipewire-0:ro`.

How is this better than using the existing jack shared-module?

**Comment by tinywrkb** [comment]
> > Look how I do this [here](https://github.com/tinywrkb/flatpaks/blob/master/shared-modules/pipewire/pipewire-jack-runtime.json). The socket is `- --filesystem=xdg-run/pipewire-0:ro`.
> 
> How is this better than using the existing jack shared-module?

Why would you compile and install a library just to remove it?  
Why would you link against a library that won't exist in the same location?  

Yes, ABI is stable and the dynamic loader does its magic, so what? PipeWire's libjack's which you gonna use anyway is already available in the runtime, and we are only missing the headers and pkgconfig file.

Anyway, it doesn't matter much, both solutions gonna be thrown soon into the trash. We just waiting for the runtime to update PipeWire to 0.3.26 and enable jack-devel.

**Comment by fastrizwaan** [comment]
Kindly merge if there are no issues with the manifest and appdata...

**Comment by hfiguiere** [comment]
bot, build io.mpv.Mpv

**Comment by fastrizwaan** [comment]
bot, build io.mpv.Mpv

**Comment by fastrizwaan** [comment]
> /merge

Thanks a million. Appreciate it.

**Comment by PranavBhattarai** [comment]
@fastrizwaan I hope I'm not asking the wrong question, but when user will be able to do this: `flatpak install mpv`

**Comment by fastrizwaan** [comment]
> @fastrizwaan I hope I'm not asking the wrong question, but when user will be able to do this: `flatpak install mpv`

right now! it's published on flathub. https://flathub.org/apps/details/io.mpv.Mpv

**Comment by PranavBhattarai** [comment]
@fastrizwaan Thank you so much for doing this.

**Comment by barthalion** [comment]
Side-note: I think this ID is just wrong and should be all lowercase to follow the website description. While it ultimately doesn't matter, I wish it was paid more attention to.

**Comment by fastrizwaan** [comment]
> Side-note: I think this ID is just wrong and should be all lowercase to follow the website description. While it ultimately doesn't matter, I wish it was paid more attention to.

Shall we change `app-id: io.mpv.mpv ` ; if possible could you please do it and re-merge (if required)

**Comment by barthalion** [comment]
At this point it's already too late.

**Pull request closed without merging**